plugins/TTAnalyzer.cc

```diff
@@ -5,104 +7,449 @@
+    // Always take care that undefined quantities can still be accessed => it will be the categories' role
+    // to determine whether a quantity is well-defined or not.
+
+    for(const std::string &wp: {"LL", "MM", "TT"}){
+      // Pt-ordered
+      diBJet_p4[wp + "-Pt"] = m_diBJet_PtChosen[wp].p4;
+      diBJet_idxs[wp + "-Pt"] = std::make_pair(m_diBJet_PtChosen[wp].idxs.first, m_diBJet_PtChosen[wp].idxs.second);
+      if(m_diBJet_PtChosen[wp].idxs.first >=0 ){
+        diBJet_DR[wp + "-Pt"] = VectorUtil::DeltaR(jets.p4[m_diBJet_PtChosen[wp].idxs.first], jets.p4[m_diBJet_PtChosen[wp].idxs.second]);
+        diBJet_DPhi[wp + "-Pt"] = VectorUtil::DeltaPhi(jets.p4[m_diBJet_PtChosen[wp].idxs.first], jets.p4[m_diBJet_PtChosen[wp].idxs.second]);
+        diBJet_DEta[wp + "-Pt"] = DeltaEta(jets.p4[m_diBJet_PtChosen[wp].idxs.first], jets.p4[m_diBJet_PtChosen[wp].idxs.second]);
+      }else{
+        diBJet_DR[wp + "-Pt"] = 0;
+        diBJet_DPhi[wp + "-Pt"] = 0;
+        diBJet_DEta[wp + "-Pt"] = 0;
+      }
+      selectedJets_diBJetExcluded[wp + "-Pt"] = std::vector<uint8_t>();
+      for(uint8_t ijet = 0; ijet < selectedJets_tightID_DRcut.size(); ijet++){
+        if(m_diBJet_PtChosen[wp].idxs.first != ijet && m_diBJet_PtChosen[wp].idxs.second != ijet)
+          selectedJets_diBJetExcluded[wp + "-Pt"].push_back(ijet);
+      }
+
+      // CSVv2-ordered
+      diBJet_p4[wp + "-CSVv2"] = m_diBJet_CSVv2Chosen[wp].p4;
+      diBJet_idxs[wp + "-CSVv2"] = std::make_pair(m_diBJet_CSVv2Chosen[wp].idxs.first, m_diBJet_CSVv2Chosen[wp].idxs.second);
+      if(m_diBJet_CSVv2Chosen[wp].idxs.first >=0 ){
+        diBJet_DR[wp + "-CSVv2"] = VectorUtil::DeltaR(jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.first], jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.second]);
+        diBJet_DPhi[wp + "-CSVv2"] = VectorUtil::DeltaPhi(jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.first], jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.second]);
+        diBJet_DEta[wp + "-CSVv2"] = DeltaEta(jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.first], jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.second]);
+      }else{
+        diBJet_DR[wp + "-CSVv2"] = 0;
+        diBJet_DPhi[wp + "-CSVv2"] = 0;
+        diBJet_DEta[wp + "-CSVv2"] = 0;
+      }
+      selectedJets_diBJetExcluded[wp + "-CSVv2"] = std::vector<uint8_t>();
+      for(uint8_t ijet = 0; ijet < selectedJets_tightID_DRcut.size(); ijet++){
+        if(m_diBJet_CSVv2Chosen[wp].idxs.first != ijet && m_diBJet_CSVv2Chosen[wp].idxs.second != ijet)
+          selectedJets_diBJetExcluded[wp + "-CSVv2"].push_back(ijet);
+      }
+
+    }
+
+    ///////////////////////////
+    //    EVENT VARIABLES    //
+    ///////////////////////////
+    
+    // leptons-jets
+   
+    ll_jj_p4 = diLepton_p4 + diJet_p4;
+    ll_jj_DR = VectorUtil::DeltaR(diLepton_p4, diJet_p4);
+    ll_jj_DPhi = VectorUtil::DeltaPhi(diLepton_p4, diJet_p4);
+    ll_jj_DEta = DeltaEta(diLepton_p4, diJet_p4);
+
+    lj_minDR = 10;
```

Comment:
I prefer to use `std::numeric_limits<TYPE>::max()` instead of hardcoding some value